platform/cosmos/model.go

```diff
@@ -0,0 +1,46 @@
+package cosmos
+
+// Tx - Base transaction object. Always returned as part of an array
+type Tx struct {
+	Block  string `json:"height"`
+	Date   string `json:"timestamp"`
+	ID     string `json:"txhash"`
+	TxData TxData `json:"tx"`
+}
+
+// TxData - "tx" sub object
+type TxData struct {
+	TxType     string     `json:"type"`
+	TxContents TxContents `json:"value"`
+}
+
+// TxContents - amount, fee, and memo
+type TxContents struct {
+	TxMessage []TxMessage `json:"msg"`
+	TxFee     TxFee       `json:"fee"`
+	TxMemo    string      `json:"memo"`
+}
+
+// TxMessage - an array that holds multiple 'particulars' entries. Possibly used for multiple transfers in one transaction?
+type TxMessage struct {
+	TxParticulars TxParticulars `json:"value"`
+}
+
+// TxParticulars - from, to, and amount
+type TxParticulars struct {
+	FromAddr string   `json:"from_address"`
+	ToAddr   string   `json:"to_address"`
+	TxAmount []Amount `json:"amount"`
+}
+
+// TxFee - also references the "amount" struct
+type TxFee struct {
+	FeeAmount []Amount `json:"amount"`
+}
+
+// Amount - the asset & quantity. Always seems to be enclosed in an array/list for some reason.
+// Perhaps used for multiple tokens transferred in a single sender/reciever transfer?
```

Comment:
Can you take a look in the Cosmos docs? 90% sure though that it's just going to be one item

Reply:
Will do

Reply:
I couldn't find anything in the docs about it - shall I hardcode it to expect only one array item?